lib/src/main/java/co/nimblehq/recentapps/thumbnailhiding/navbar/NavigationBarObserver.java

```diff
@@ -0,0 +1,134 @@
+package co.nimblehq.recentapps.thumbnailhiding.navbar;
+
+import android.content.Context;
+import android.database.ContentObserver;
+import android.net.Uri;
+import android.os.Build;
+import android.os.Handler;
+import android.os.Looper;
+import android.provider.Settings;
+
+import java.util.ArrayList;
+
+/**
+ * copy from https://github.com/gyf-dev/ImmersionBar.
+ * 导航栏显示隐藏处理，目前只支持emui和miui带有导航栏的手机
+ *
+ * @author geyifeng
+ * @date 2019/4/10 6:02 PM
+ */
+public final class NavigationBarObserver extends ContentObserver {
```

Comment:
Optional: Re-factor `NavigationBarObserver`, `OSUtils` &`OnNavigationBarListener`  to Kotlin

Reply:
Possible, but I would like to keep them in java from the original source for further updates 🙏